src/averageAges.js

```diff
@@ -55,7 +67,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const difference = (!onlyWithSon)
+    ? people.filter(child =>
+      people.find(mother => child.mother === mother.name))
+    : people.filter(child =>
+      people.find(mother => child.mother === mother.name) && child.sex === 'm');
+
+  const averageAgeDiff = difference.reduce((total, child) =>
+    total + (child.born - people.find(mother =>
+      child.mother === mother.name).born), 0) / difference.length;
```

Comment:
[checklist point 2](https://github.com/mate-academy/js_average-ages/blob/master/checklist.md)

It would be more clear to create `mother` variable.